### ISSUE TITLE
[css-scroll-snap-1] Should include example from the demise of css-sticky-scrollbars

### ISSUE THREAD

**Issue by SamB** [w3c/csswg-drafts#4371]
<https://drafts.csswg.org/css-scroll-snap-1/#examples> doesn't seem to have anything like the example in <http://tabatkins.github.io/specs/css-sticky-scrollbars/#intro> (used to demonstrate that we don't *need* a sticky scrollbars spec, because scroll snap can already do that).

Here, I'll just quote the relevant portion:

<blockquote>
   <p>This spec has been superseded by other CSS technology.
In particular, to implement a chatroom
that "sticks" the scrollbar to the bottom of the scroller when you’re near the bottom,
staying there if you add new messages to the bottom,
but not messing with your scroll position when you’re scrolled elsewhere in the history,
just use <a data-link-type="biblio" href="">[css-scroll-snap-1]</a>:</p>
<pre lang="css">
.message-container {
  scroll-snap-type: proximity;
}
.message-container::after {
  content: "";
  height: 0;
  overflow: hidden;
  display: block;
  scroll-snap-align: end;
}
</pre>
   <p>This creates a single <a data-link-type="dfn" href="https://drafts.csswg.org/css-scroll-snap-1/#scroll-snap-area" id="ref-for-scroll-snap-area">scroll snap area</a> in the message container,
aligned with the very bottom of the container.
If you scroll "near" the bottom, you’ll <a data-link-type="dfn" href="https://drafts.csswg.org/css-scroll-snap-1/#snap" id="ref-for-snap">snap</a> to it;
if you add more content to the message container
(thus pushing the ::after further down),
it’ll automatically re-snap to it
(because <a data-link-type="dfn" href="https://drafts.csswg.org/css-overflow-3/#scroll-container" id="ref-for-scroll-container">scroll containers</a> have to re-snap to the same <span>scroll snap area</span> if it still exists);
if you are scrolled somewhere else in the history,
it won’t do anything at all.</p>
</blockquote>

**Comment by majido** [comment]
Note that I think to have the css-sticky-scrollbar functionality implementation need to re-snap after layout changes. 

As far as I am aware no implementation is doing so correctly despite it being in the specification. Here is a [test page](https://snap.glitch.me/snap-after-layout.html) I created to show case this. Note that [`scroll-anchoring`](https://drafts.csswg.org/css-scroll-anchoring-1/) does emulate the re-snap behavior to some extent so I had to disable it to make the effect more pronounced. [Testing](https://snap.glitch.me/snap-after-layout.html ) this in Firefox and Chrome shows that no one is doing the correct thing yet.

Note that we are working toward implementing this in Blink ([bug](https://bugs.chromium.org/p/chromium/issues/detail?id=866127)) but I cannot provide when that will land.

**Comment by tabatkins** [comment]
Yes, the resnap is def required per spec and needed to get the referenced behavior. (It's one of the reasons we have that requirement in the spec!)

**Comment by fantasai** [comment]
I had thought we intended it to be more easily solved by https://www.w3.org/TR/css-align-3/#overflow-scroll-position ...

Anyway, I added the example to the spec. Also a subheading for the section so it's easier to find.
